marketdata/stream/client.go

```diff
@@ -440,10 +459,34 @@ func (c *client) maintainConnection(ctx context.Context, u url.URL, initialResul
 			} else {
 				c.logger.Warnf("datav2stream: connection lost")
 			}
+
+			if c.disconnectCallback != nil {
+				callbackWaitGroup.Add(1)
+				go func() {
+					defer callbackWaitGroup.Done()
+					c.disconnectCallback()
+				}()
+			}
 		}
 	}
 }
 
+// waitTimeout waits for the WaitGroup for the specified max timeout.
+// Returns true if waiting timed out.
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
```

Comment:
[minor]: I would perhaps rename this to `waitWithTimeout` and return false when there's a timeout instead of the other way around.
Both of these are minor comments so feel free to ignore them if you think the current solution is better!